scripts/macos/agent_diagnostics.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+# agent_diagnostics.sh collects all artifacts required for troubleshooting,
+# including logs, configuration files, and application files. Those artifacts
+# are compressed into a zip file, along with an output file of relevant information
+# about the system for quick referrence and an inventory of the files included
+# in the zip file.
+
+if [[ "${UID}" != 0 ]]; then
+  (>&2 echo "Error:  $0 must be run as root")
+  exit 1
+fi
+
+# Some global variables
+JCPATH="/opt/jc"
+JCLOG="/var/log/"
+STAMP=$( date +"%Y%m%d%H%M%S" )
+ZIPFILE="./jc${STAMP}.zip"
+declare -a INVENTORY
+
+# Is jcagent installed? if not, exit.
+if [[ ! -d "${JCPATH}" ]]; then
+  echo "JCAGENT IS NOT INSTALLED ON THIS MACHINE."
+  exit 1
+fi
+
+# Is zip installed?
+if ! which zip 2> /dev/null; then
+  ZPATH="false"
+else
+  ZPATH=$(which zip)
+fi
+
+function indent() {
+	sed 's/^/		/g'
+}
+
+function zipjc() {
+  # Take inventory of files to be zipped.
+  for i in "${JCPATH}"/*; do
+    INVENTORY+=("${i}")
+  done
+
+  # check to see if zip exists.
+  if [[ "$ZPATH" = "false" ]]; then
+    ZIPIT="zip is not installed. please send the following files with your support request:\n${INVENTORY[*]}"
+  else
+    if [[ -f "${ZIPFILE}" ]]; then
+      mv "${ZIPFILE}" ./jc"${STAMP}".bak.zip
+      zip -r "*.crt" -r "${ZIPFILE}" "${JCPATH}" > /dev/null 1
```

Comment:
As with the Linux script, this should be `zip -x "*.crt" -r "${ZIPFILE}" "${JCPATH}" > /dev/null 1`.

Reply:
Furthermore, we should exclude the `.key` files as we do in the Windows script.

Reply:
I reworked this part to perform the exclusions when generating the inventory, instead of when updating the zip file. This change will be made in the linux script as well.

Reply:
I reworked the ziplog function to test for the presence of the jumpcloud-loginwindow directory for inclusion in the zip archive.